x-pack/plugins/endpoint/common/types.ts

```diff
@@ -115,6 +119,50 @@ export interface EndpointMetadata {
   };
 }
 
+export interface ResolverPhase0Data {
+  endgame: {
+    event_type_full: string;
+    event_subtype_full: string;
+    unique_pid: number;
+    unique_ppid: number;
+  };
+  agent: {
+    id: string;
+  };
+}
+
+export interface ResolverPhase1Data {
+  event: {
+    category: string;
+    type: string;
+  };
+  endpoint: {
+    process: {
+      entity_id: string;
+      parent: {
+        entity_id: string;
+      };
+    };
+  };
+}
+
+export type ResolverData = ResolverPhase0Data | ResolverPhase1Data;
```

Comment:
Why not version? I am guessing we are caring over the business language into the code. What happens when we stop saying phases and we are mature.

Reply:
How about `ResolverLegacyData` => evrelay SMP data, and `ResolverElasticData` for the new format? @james-elastic @oatkiller @andrewstucki

Reply:
👍 "phase X" should not be referenced in our code... only makes sense to us, but future code maintainers will have no idea what it means.

Reply:
I'm also hesitant about versioning this stuff directly, mainly because we're really working  with two different datasets, one that isn't fully conformant to the ECS scheme we're going forward with (and hence requires some logic highly specific to our old payloads), and one which is (and can thus be used for data sources other than the Endpoint)--so it's almost less of a `v0` v. `v1` and more of a `v -1` v. `v1` -- I do see a benefit in versioning the ECS-compatible stuff though.

Reply:
All right I'm going with changes from `phase0` to `legacy` and `phase1` to `ElasticEndpoint`